docs/tutorials/deploy_openfaas_workload.md

```diff
@@ -0,0 +1,291 @@
+
+After Rudr was installed, we need to install [OpenFaaS Operator](https://github.com/openfaas-incubator/openfaas-operator), that will be the real workload executor.
+
+```shell script
+# create OpenFaaS namespaces
+kubectl apply -f https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml
+
+# add OpenFaaS Helm repo
+helm repo add openfaas https://openfaas.github.io/faas-netes/
+
+# generate a random password
+PASSWORD=$(head -c 12 /dev/urandom | shasum| cut -d' ' -f1)
+
+kubectl -n openfaas create secret generic basic-auth \
+--from-literal=basic-auth-user=admin \
+--from-literal=basic-auth-password="$PASSWORD"
+
+# get latest chart version
+helm repo update
+
+# install OpenFaaS operator with basic auth enabled
+# functionNamespace should be changed to the namespace in which Rudr is installed, in our case, it's default namespace
+helm install openfaas openfaas/openfaas \
+    --namespace openfaas  \
+    --set basic_auth=true \
+    --set functionNamespace=default \
+    --set operator.create=true
+```
+
+If you have trouble connecting with the chart repository, just Clone the [faas-netes](https://github.com/openfaas/faas-netes) repository and install:
+
+```
+git clone https://github.com/openfaas/faas-netes.git
+helm install openfaas faas-netes/chart/openfaas \
+    --namespace openfaas  \
+    --set basic_auth=true \
+    --set functionNamespace=default \
+    --set operator.create=true
+```
+
+Finally you could verify the installation by checking the pods.
+
+```shell script
+$ kubectl -n openfaas get pods
+NAME                                 READY   STATUS    RESTARTS   AGE
+alertmanager-67dd8444ff-nn4k8        1/1     Running   0          4m42s
+basic-auth-plugin-6f5878475f-ln2k2   1/1     Running   0          4m42s
+faas-idler-5c87fbcbdf-tsscj          1/1     Running   2          4m42s
+gateway-559fb6f5b8-bsdzq             2/2     Running   0          4m42s
+nats-79ffb4bb68-vgzct                1/1     Running   0          4m42s
+prometheus-74bc4d6884-rlj2f          1/1     Running   0          4m42s
+queue-worker-55fd5749d-7q678         1/1     Running   1          4m42s
+```
+
+Fetch your public IP or NodePort via `kubectl get svc -n openfaas gateway-external -o wide`.
+
+If using a remote cluster, you can port-forward the gateway to your local machine:
+
+```shell script
+kubectl port-forward -n openfaas svc/gateway 31112:8080 &
+```
+
+Then you could visit the OpenFaaS web by `http://127.0.0.1:31112`.
+
+User name is `admin`, and you could get the password by:
+
+```shell script
+$ echo $PASSWORD
+1b26c16c5af2343ee82be46635c0b44f7457a176
+```
+
+
+## Create Component with OpenFaaS Workload
+
+An OpenFaaS workload example is like below:
+
+```yaml
+apiVersion: core.oam.dev/v1alpha1
+kind: ComponentSchematic
+metadata:
+  name: openfaas
+spec:
+  workloadType: openfaas.com/v1alpha2.Function
+  osType: linux
+  parameters:
+    - name: image
+      type: string
+      default: alexellis/gofast023
```

Comment:
I recommend using this which is newer https://github.com/alexellis/echo-fn

Reply:
do you mean using this image? `alexellisuk/go-echo:0.2.2`